### PR TITLE
feat: LSP hover now includes "Go to" links

### PR DESCRIPTION
**Pull request by asterite**
# Description

## Problem

In Rust LSP, when you hover over a variable you get to see its type, but you also get links to every reachable type. I wanted to do that when implementing hover but I didn't know how. Now I figured it out :-)

## Summary

This is what this PR does:

https://github.com/user-attachments/assets/7362d51d-c836-4ab1-b840-d2e7f3fb38d4

## Additional Context

None.

## Documentation

Check one:
- [x] No documentation needed.
- [ ] Documentation included in this PR.
- [ ] **[For Experimental Features]** Documentation to be submitted in a separate PR.

# PR Checklist

- [x] I have tested the changes locally.
- [ ] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

**Review by jfecher — APPROVED** [review]
LGTM

File: tooling/lsp/src/requests/hover.rs [hunk 1]
```diff
@@ -361,6 +369,144 @@ fn format_parent_module_from_module_id(
...
+        match var {
+            TypeBinding::Bound(typ) => {
+                self.gather_type_links(typ);
+            }
+            TypeBinding::Unbound(..) => (),
+        }
+    }
+
+    fn push_link(&mut self, link: String) {
+        if !self.links.contains(&link) {
```
Comment (michaeljklein) [inline tooling/lsp/src/requests/hover.rs:492]: Would it make more sense to use a `HashSet` for `links` to avoid this search?

**Review by michaeljklein — CHANGES_REQUESTED** [review]

File: tooling/lsp/src/requests/hover.rs (hunk 1 above)
Comment (jfecher) [reply tooling/lsp/src/requests/hover.rs:492]: I saw this too but figured for the small amount of links within most types it wouldn't make much of a difference

Comment (asterite) [reply tooling/lsp/src/requests/hover.rs:492]: I thought about it, but I wanted the links to preserve insertion order, so if you have `Map<Key, Value>` you'd get links for `Map | Key | Value`.

Given that usually we'll end up with just a few links (usually 2 or 3) doing a linear scan is probably good enough. Alternatively we could use a crate for a dictionary that preserves order... but I didn't know if it was worth it.

**Review by michaeljklein — APPROVED** [review]

**Pull request merged**